### ISSUE TITLE
[$2000] Pressing `space` 2 times `.` appears in display name

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17153]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to iOS settings > iPhone Settings >> General >> Keyboards > toggle on "." Shortcut 
2. Open the ios mobile app or iOS Safari or Mac Desktop Chrome
3. Tap Setting > Profile
4. Tap Display Name remove name
5. Write new text and press space button 2 times and see the result (the "." shortcut doesn't work even though it's enabled)

## Expected Result:
if "." shortcut is enabled, the shortcut should work when tapping the space 2 times

## Actual Result:
if "." shorcut is enabled, the shortcut doesn't work when tapping the space 2 times

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.96-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/230680037-97e38841-3890-4671-a905-6757633be3f0.mp4

https://user-images.githubusercontent.com/43996225/230680041-a8b718bc-2575-43d1-9d83-dc2662c3094a.mp4

https://user-images.githubusercontent.com/43996225/230680087-d94f2354-2a45-498a-9705-f9f691ea13e3.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @harshad2711
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1680851400066889

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0194127eb40345b2ea</li>
        <li>Upwork Job ID: 1646595046183731200</li>
        <li>Last Price Increase: 2023-04-20</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @gedu</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Christinadobrzyn (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by PrashantMangukiya** [comment]
This is not issue.  This is iOS keyboard specific features.

Follow this Steps to Disable It: 
Go to **iPhone Settings >> General >> Keyboards >>**

You will see this screen, So disable **"." Shortcut** settings as show below, It will solve the problem.
![IMG_5909](https://user-images.githubusercontent.com/7823358/230723444-ed9b6108-4508-41a1-b9ec-8b573e035295.PNG)

**Comment by MelvinBot** [comment]
@Christinadobrzyn Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by Christinadobrzyn** [comment]
Thanks @PrashantMangukiya! I'll reach out in the slack chat to see if we can do some more testing - https://expensify.slack.com/archives/C049HHMV9SM/p1681247093995989?thread_ts=1680851400.066889&cid=C049HHMV9SM

**Comment by Christinadobrzyn** [comment]
did some testing and updated the OP - asking for a quick review to make sure I understand correctly - https://expensify.slack.com/archives/C049HHMV9SM/p1681334171519149?thread_ts=1680851400.066889&cid=C049HHMV9SM

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0194127eb40345b2ea

**Comment by MelvinBot** [comment]
Current assignee @Christinadobrzyn is eligible for the External assigner, not assigning anyone new.

**Comment by Christinadobrzyn** [comment]
Updated OP based on Slack [chat](https://expensify.slack.com/archives/C049HHMV9SM/p1681334171519149?thread_ts=1680851400.066889&cid=C049HHMV9SM) - I think this can be External.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @0xmiroslav (`External`)

**Comment by MelvinBot** [comment]
@chiragsalian, @Christinadobrzyn, @0xmiroslav Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by hasebsiddiqui** [comment]
https://github.com/facebook/react-native/issues/27693

**Comment by Christinadobrzyn** [comment]
Looks like we're waiting on more proposals

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $2000

**Comment by Christinadobrzyn** [comment]
Raised price to $2000

**Comment by PrashantMangukiya** [comment]
@Christinadobrzyn  @chiragsalian  @0xmiroslav  I think this is working as expected in our app and there is no issue.

**Here is what happen,** 
When you tap space twice at the end of sentence it will put "." 
_Now delete that dot and tap space twice again it will not put dot._

Add some text and tap space twice it will put "."
_Now delete that dot and tap space twice again it will not put dot._

Add some text and tap space twice it will put "."
_Now delete that dot and tap space twice again it will not put dot._

Repeat this step, it will show same behaviour. 

I tested it within message app. This is expected behaviour,  it looks like there is not any issue in our app. Below is the video for comparison.

### Expensify App
https://user-images.githubusercontent.com/7823358/233493871-7b6aceaf-c00f-46b1-a684-dda02288b526.mov

### Message App
https://user-images.githubusercontent.com/7823358/233494889-9c5bf7b4-164e-4563-98a9-0f1690427c98.mov

Thank you.

**Comment by Christinadobrzyn** [comment]
Thanks for reviewing @PrashantMangukiya! To confirm, do you have "." enabled or disabled in the iOS settings?

**Comment by PrashantMangukiya** [comment]
> Thanks for reviewing @PrashantMangukiya! To confirm, do you have "." enabled or disabled in the iOS settings?

@Christinadobrzyn  Yes "." enabled in the iOS keyboards settings.

**Comment by Christinadobrzyn** [comment]
ah okay! I'll close this for now. Thank you so much for your help looking into this!

**Comment by PrashantMangukiya** [comment]
@Christinadobrzyn I think what @harshad2711 said in above [comment](https://github.com/Expensify/App/issues/17153#issuecomment-1517210305) is correct for iOS native. 

In fact what I noticed is, in iOS Safari it shows dot upon double space. But in iOS Native it is does not show, wired behaviour. i.e. three dot works on iOS Safari but not iOS native.   **I think we should re-open issue to make it working for iOS Native also.**  Sorry for the confusion.

####  iOS Native (Dot Not Working)

https://user-images.githubusercontent.com/7823358/233536734-ad486aad-18ca-491f-b86a-fd6e312959bf.MP4

#### iOS Safari (Dot Working)

https://user-images.githubusercontent.com/7823358/233536839-8ff4a215-a16a-4eb2-b9f2-99acb04dd88b.MP4

**Comment by chiragsalian** [comment]
Yup reopening, looks like a legit bug

**Comment by MelvinBot** [comment]
@chiragsalian, @Christinadobrzyn, @0xmiroslav Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by Christinadobrzyn** [comment]
Looks like we're waiting on some proposals.

**Comment by mohamedwshams** [comment]
A couple of questions/comments on this issue: 

- I tried to reproduce on iOS app and iOS Safari and issue did not exist. The iOS app and web app behave as expected when the "." option is enabled and when it on an iPhone 13 Pro Max running the latest version of iOS

- Why is this particular "bug" worth 2K USD? If it actually exists, it is not critical to the business imho

I recommend either defining when exactly this bug happens and on which device and iOS version or closing this ticket

**Comment by MelvinBot** [comment]
📣 @mohamedwshams! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by gedu** [comment]
Hey, hey, I am from Callstack - expert contributor group, I can take a look

**Comment by MelvinBot** [comment]
📣 @gedu You have been assigned to this job by @chiragsalian!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by chiragsalian** [comment]
Assigned the issue over to you @gedu

**Comment by gedu** [comment]
Still looking into this, my current findings are:

Seems that if we don't use the `value` prop from TextInput, the double-tapping space works, it adds the dot.

Here I start debugging the Bridge and if I don't use the `value` from Native it sends the `.` after the second space
 
<img width="741" alt="textInput_no_value_prop" src="https://user-images.githubusercontent.com/1676818/234816003-be17a43f-7372-4fff-adfa-37ad0118878d.png">

But when I add the `value` prop, Native doesn't send the `.` after the second space.

<img width="749" alt="textInput_value_prop" src="https://user-images.githubusercontent.com/1676818/234816249-de32c4d7-05eb-4222-b3e7-74312d9e2c0d.png">

I will need to dig deeper, I will start looking at react-native `TextInput` source code and see what happens with the `value` prop at that level.

**Comment by gedu** [comment]
Didn't have the time today to take a deeper look, I'm almost finishing the other task, as soon as I finished I continue with this one

**Comment by Christinadobrzyn** [comment]
Still investigating, please feel free to provide more of an update when you have one @gedu!

**Comment by gedu** [comment]
Tomorrow will make some time to look at this

**Comment by Christinadobrzyn** [comment]
Not sure what we need to do next for this to stay on schedule. @0xmiroslav or @chiragsalian would you have some insights into what to do next?

**Comment by gedu** [comment]
Hey, still digging into this, because adding `multiline` to all our inputs I don't think is a good solution, I wanna track what are the differences between `RCTSinglelineTextInputView` and `RCTMultilineTextInputView` so maybe I can come up with a final proposal.

**Comment by gedu** [comment]
Hey, trying to wrap another task, as soon as I finish I will continue here

**Comment by gedu** [comment]
Hey, still couldn't find the time to take a look, I will make some time tomorrow

**Comment by gedu** [comment]
Hey today I've been checking more about how the native inputs behave on iOS, Still couldn't find the right path, I will start doing a flow chart to see in which steps Multiline and Single line differ and also create an issue on react-native GH so I can get some inputs from them and probably they can guide me.

**Comment by gedu** [comment]
Looking into this a bit further, it could be something related to the autocorrect but not sure, so I created an issue on react-native: https://github.com/facebook/react-native/issues/37437

This will take longer than expected, can we change the label to weekly? @Christinadobrzyn

**Comment by strepanier03** [comment]
👋 hey, quick note that we should keep all e/app GHs as daily unless they have "Hold" in the title. I'm going to flip this back to daily as I'm doing the resync and it's [step in the SO process guide](https://stackoverflowteams.com/c/expensify/questions/8208/14658#14658).

**Comment by gedu** [comment]
I did some deeper investigation and created an Objective-C project to test the different behaviors, I could set and understand better the flow of the UITextField used by Singleline component. Also trying to get in touch with some devs in Meta/FB so I can get my issue reviewed or get some help

**Comment by gedu** [comment]
I run some tests, and seems to be working, I'm discussing a proposal internally

**Comment by gedu** [comment]
@0xmiroslav 

## Proposal

### Please restate the problem we are addressing in this issue:
The issue is that on iOS native apps, the double-tapping feature, which should add a dot when tapping the space key twice, is not working as expected.
Identifying the root cause of the problem:
The “TextInput” component on the iOS side is managed by the `RCTSinglelineTextInputView` module ( multiline=false ). When the value prop is not used, the component behaves as an uncontrolled component, where the system handles the text updates. However, when the value prop is used, the component becomes a controlled component, relying on the provided value to update the text. This controlled component setup involves increased communication between the JS and Native sides.
To ensure the native state remains synchronized, there is a method called `updateState` that handles updating the native event after a text change. However, in the case of a controlled component, the additional back-and-forth interaction between JavaScript and Native can cause the “updateState” method to be executed multiple times. During these multiple updates, the text may be replaced, even with the same text content. This discrepancy in the number of state updates disrupts the expected behavior of the double-tapping feature, which relies on the detection of consecutive spaces.

### Proposed changes to resolve the problem:
To resolve this issue, I propose adding a flag named `_stateUpdated` to control the execution of the “updateState” method. By setting this flag to ensure that “updateState” is executed only once, this will align the behavior with that of an uncontrolled component. The flag will be cleared after any new text change in the `textField:(_unused UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string` method. Since this method is always executed before “updateState” and not affected by the back-and-forth sync between JS and Native realms, it provides a reliable point to control the state updates.

This mean a fix at react-native framework

### Exploration of alternative solutions (optional):
During the investigation, I explored a couple of alternative solutions to address the issue:
- Using the `multiline` prop: This approach involves utilizing a different component that is always uncontrolled and handled by the system. However, adopting this solution may introduce challenges in making the input behave as a single-line component, potentially conflicting with the desired functionality.
- Not using the `value` prop (yes using `onChange`): By not using the `value` prop, we can revert the component to an uncontrolled state, allowing the system to handle all text changes. However, this would require accessing the text through a ref or storing it in the component’s state. While this approach may resolve the issue, it may result in the loss of text formatting features.

**Comment by chiragsalian** [comment]
Nice, great work gedu. I'll demote the issue and place HOLD on it so that melvin doesn't make it overdue each day. Feel free to remove HOLD and make issue daily once upstream issue is live.

**Comment by gedu** [comment]
Sounds great, will take care another task for a couple of days and come back to create a Draft PR on react-native framework

**Comment by gedu** [comment]
Working on another tickets that has more priority, as soon as I can have them in a good position will continue here

**Comment by gedu** [comment]
This week or next top I will start working this issue

**Comment by Christinadobrzyn** [comment]
still on hold

**Comment by gedu** [comment]
Still on hold, will try to take this bug as soon as possible, writting some big features documentation

**Comment by gedu** [comment]
Still on hold

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo until August 31st so going to unassign and assign a new teammate. 

At this time, we're on hold

**Comment by tjferriss** [comment]
Still on hold

**Comment by tjferriss** [comment]
Still on hold, MelvinBot.

**Comment by Christinadobrzyn** [comment]
back from ooo - thanks for handling this @tjferriss - I can take it off your plate.

Still on hold for https://github.com/facebook/react-native/issues/37437 - going to move this to weekly while we wait

**Comment by tjferriss** [comment]
Sounds good, thanks

**Comment by gedu** [comment]
Hey, my Mac broke a couple of weeks ago, had most of the code there, I  will have it back in 2 weeks, so I can keep working on this issue, sorry for the delay

**Comment by Christinadobrzyn** [comment]
Hi @gedu - do you have an update on this?

**Comment by gedu** [comment]
Next week I will have my Mac back and can start again with this one, sorry for the delay

**Comment by gedu** [comment]
I have my Mac back, I've been setting up so it is in good shape, so beginning of next week I can keep working on this. I had a solid solution I was mainly testing so I didn't break anything else
@chiragsalian

**Comment by Christinadobrzyn** [comment]
Hi @gedu do you have an update on this for us?

**Comment by gedu** [comment]
Hey, I finished a couple of tickets, today I start again to look into this, I will check were I left everything and continue doing any changes, updates and let you know

**Comment by gedu** [comment]
I created a PR for this issue on react-native repository: https://github.com/facebook/react-native/pull/39385

**Comment by gedu** [comment]
Working on the PR, I got asked to add some integration tests, so working on it

**Comment by gedu** [comment]
Started with integration tests for the InputText

**Comment by Christinadobrzyn** [comment]
Hi @gedu can you add an update here for us? Thanks!

**Comment by Christinadobrzyn** [comment]
Hi @gedu just checking on this - can you leave an update when possible?

**Comment by gedu** [comment]
Hey, sorry I had to jump among other tickets, I updated the code and seems there were a lot of changes, making my fix not work anymore, so I need to investigate again.
Left a comment a couple of days ago https://github.com/facebook/react-native/pull/39385#issuecomment-1767657525. Will try to resume as soon as possible but the next weeks will be kind of complicated for me given that I have to work on another feature.

**Comment by Christinadobrzyn** [comment]
Hi @gedu do you have any update on this job? Thanks!

**Comment by Christinadobrzyn** [comment]
Hi @gedu please let us know of an update when you can!

**Comment by Christinadobrzyn** [comment]
asking in Slack for an update to this GH - https://expensify.slack.com/archives/C03UK30EA1Z/p1699661800466839

**Comment by gedu** [comment]
Hey, sorry, I didn't have the time to resume this bug, I will try to dedicate some time this week, had to focus on other features.

**Comment by chiragsalian** [comment]
any update here @gedu?

**Comment by gedu** [comment]
Still working on this

**Comment by gedu** [comment]
Was trying to find a fix with fabric=false, but I couldn't find a clear pattern, but since fabric is true by default, I'm trying to find a fix just for fabric enabled

**Comment by gedu** [comment]
I found the fix, not that different as the last time, but seems that it only works with fabric, I'm double checking and creating integration tests

**Comment by Christinadobrzyn** [comment]
Hi @harshad2711 I can help with that question! Reporters are paid when the job is complete and it's been 7 days in production (details in our [guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md))That's the standard payment timeframe for all participants (reporters, contributors, C+) on a job. Let me know if you have any questions!

**Comment by gedu** [comment]
Integration tests are working on Android, but not on iOS, seems that webdriver or appium isn't getting the text from the Input, looking into it

**Comment by gedu** [comment]
Integration tests working for Android and iOS, pinged the reviewer so he can take a look

**Comment by gedu** [comment]
Pinging the reviewer of the PR and also chatting with [another dev](https://github.com/facebook/react-native/issues/37437) that works with Text and TextInput

**Comment by gedu** [comment]
@0xmiroslav @mountiny Probably I will need some backup, so I can get the PR reviewed https://github.com/facebook/react-native/pull/39385

**Comment by mallenexpensify** [comment]
@eVoloshchak reassigning, please take over as C+. If you don't have bandwidth, unassign yourself. Thanks

**Comment by eVoloshchak** [comment]
@gedu, the current status on this is we're expecting a response from FB, is that correct?

We're facing the same problem in https://github.com/Expensify/App/issues/19507#issuecomment-1894295180
> this is reliant on pull request to the react-native repository... Is there anything we (or one of our expert contributor agencies) can do to get this moving along? It's literally been months of nothing, unfortunately.

Linking here in case something can be done about it

**Comment by gedu** [comment]
@eVoloshchak Yes, it is correct

**Comment by Christinadobrzyn** [comment]
just to confirm an update here - @gedu are you working on a resolution to this or do you need us to reach out to anyone else for review/assistance?

**Comment by mountiny** [comment]
We are waiting for a response, nothing actionable right now. Any updates @gedu ?

**Comment by Christinadobrzyn** [comment]
Hi @gedu any update on the feedback? Thanks!

**Comment by gedu** [comment]
No updates, I pinged again

**Comment by gedu** [comment]
I could confirm that my PR fixes another issue, trying to get more visibility: https://github.com/facebook/react-native/pull/39385#issuecomment-1973953402

**Comment by Christinadobrzyn** [comment]
hi @gedu how are things going? If you can provide an update that'd be great! Thank you!

**Comment by gedu** [comment]
Hey hey, the PR is getting more views, I will try to get more help

**Comment by gedu** [comment]
No updates yet

**Comment by gedu** [comment]
I joined the React Discord, so I'm trying to get some help

**Comment by gedu** [comment]
With Michal Pierzchala, we placed a message into a channel (code-review), no news yet

**Comment by Christinadobrzyn** [comment]
just checking in on this @gedu. thanks!

**Comment by gedu** [comment]
I ping on the PR and on the Discord channel again, but no news

**Comment by Christinadobrzyn** [comment]
Just checking in @gedu. thank you! Do you think this needs to be moved to Monthly? cc @chiragsalian

**Comment by Christinadobrzyn** [comment]
Should we move this to monthly @chiragsalian @gedu?

**Comment by gedu** [comment]
> Should we move this to monthly @chiragsalian @gedu?

Maybe yes, I still have no responses

**Comment by gedu** [comment]
No updates, I have some conflicts on my PR that I will fix soon

**Comment by gedu** [comment]
No yet, I didn't have the time to fix the conflicts, I working on other Critical tickets

**Comment by gedu** [comment]
Keep looking if I can place my fix in another place of the code as suggested

**Comment by gedu** [comment]
I found another way to fix the bug without using a new flag for early exit, waiting for it

**Comment by gedu** [comment]
I pinged the reviewer, still waiting

**Comment by gedu** [comment]
Hey, I will ping today, still no news. I created another issue related with TextInput, if that one get attention I will try to mention this one, so maybe it gets some momentum again. 
I will be on vacation, when I'm back I will check the situation again

**Comment by Christinadobrzyn** [comment]
Hi @gedu just checking in on this for an update - thank you!

**Comment by gedu** [comment]
Hey, no yes, I pinged less of 2 weeks ago, but no response. I will try on Discord

**Comment by gedu** [comment]
Hey, I wrote on Discord trying to call my PR attention, still no news

**Comment by mallenexpensify** [comment]
Thanks @gedu , we've def found that upstream fixes can take a while.

**Comment by gedu** [comment]
I sent another ping to the reviewer

**Comment by gedu** [comment]
Posting my PR on these thread to see if it can get some visibility: https://x.com/dan_abramov2/status/1842513065952354438

**Comment by mallenexpensify** [comment]
Also noticed @mrousavy from Margelo commented there. We work with them in case that's relevant or helpful.  :)

**Comment by mrousavy** [comment]
lol, this is exactly what we talked about yesterday @hannojg.
I think we can fix this, no?

**Comment by gedu** [comment]
Seems that got some attraction, the reviewer is taking a look, I will fix the conflicts and clean up a bit the code: https://github.com/facebook/react-native/pull/39385#issuecomment-2402159663

**Comment by gedu** [comment]
I just updated the PR, solved the conflicts, tomorrow I have to fix some examples I think

**Comment by gedu** [comment]
I fixed some examples that changed after a merge conflict

**Comment by gedu** [comment]
Nick made a fix, that I think is related with mine, his handle on a [more generic way,](https://github.com/facebook/react-native/pull/39385#pullrequestreview-2370188822) we should test if this bug is still happening after his [fix](https://github.com/facebook/react-native/pull/46970), So I will close my PR

**Comment by AndrewGable** [comment]
So we will just have to update to the version of react native that contains this PR, right?

**Comment by gedu** [comment]
@AndrewGable yes

@Christinadobrzyn not sure what should be the next step for this issue

**Comment by mhoran** [comment]
Unfortunately this issue does not seem to be resolved with the patch in https://github.com/facebook/react-native/pull/39385#pullrequestreview-2370188822 as reported in https://github.com/facebook/react-native/issues/27693#issuecomment-2439826790.

I may have messed something up, but I tried using [patch-package](https://www.npmjs.com/package/patch-package) to apply the patch. The patch definitely compiles, but there is no behavior change and the bug persists in New Arch.

**Comment by mhoran** [comment]
With all the patches mentioned in https://github.com/reactwg/react-native-releases/issues/595#issuecomment-2442983461, this issue should be resolved once and for all. The pick request should result in these patches making their way into a future 0.76 release. In the meantime, you can apply the patches manually atop 0.76 as shown in the [reproducer](https://github.com/mhoran/new-arch-autocorrect-repro).

**Comment by Christinadobrzyn** [comment]
Just checking on this, I'm not sure what we should do here. @mhoran @eVoloshchak @gedu should we continue to keep this on hold for the patch?

**Comment by mhoran** [comment]
The patch landed in React Native v0.76.2. Note that the app must be moved to the New Architecture, as the fix was only implemented for New Architecture.

**Comment by mhoran** [comment]
It looks like the app is still on 0.75.2. It needs to first be updated to 0.76.2 and also migrated to New Architecture. I'm not sure if New Architecture is already enabled; but it is enabled by default in 0.76. I see a PR here to update to 0.76.3: https://github.com/Expensify/App/pull/51475 -- so when that is merged I expect this issue should be resolved (so long as New Arch is enabled.)

**Comment by WoLewicki** [comment]
New architecture in Expensify App has been enabled for about 9 months now: https://github.com/Expensify/App/pull/13767 if it is what you mean @mhoran.

**Comment by Christinadobrzyn** [comment]
Hi @mhoran - just checking again for an update on this. Thanks!

**Comment by mhoran** [comment]
I see that #51475 was merged and can confirm the issue is resolved:

![Image](https://github.com/user-attachments/assets/eef7d5b8-feb3-4804-a8cb-1d65765c557d)

**Comment by Christinadobrzyn** [comment]
Oh wow! So https://github.com/Expensify/App/pull/51475 fixed this?! 

what's next, do we need to pay anyone for this?

**Comment by mhoran** [comment]
The underlying issue was [fixed by Meta](https://github.com/facebook/react-native/pull/46970). A subsequent [patch](https://github.com/facebook/react-native/pull/47269) fixed the underlying issue for the `value` prop after I reported this was still broken in the [pick request](https://github.com/reactwg/react-native-releases/issues/595#issuecomment-2441737925). All fixes were picked to 0.76 per my [pick request](https://github.com/reactwg/react-native-releases/issues/595). https://github.com/Expensify/App/pull/51475 upgraded Expensify to 0.76, so the changes came in and the issue was fixed!

I'm not sure how the payouts work. but my guess would be no since Meta fixed this independently.

**Comment by Christinadobrzyn** [comment]
Thank you @mhoran! Feel free to provide an update when possible. Do you think we should keep this with the weekly label or are updates on this going to be closer to monthly?

**Comment by mhoran** [comment]
@Christinadobrzyn I believe this is resolved, but I'm just a random community member who stumbled across this issue. I've done all I can at this point, and am not sure what else is needed to close this out (I can't do that myself.)

**Comment by eVoloshchak** [comment]
@Christinadobrzyn, here is the current status:

- The issue is resolved* now (we should do 2 weeks of KI tests though)
- This was being fixed by @gedu in https://github.com/facebook/react-native/pull/39385 and while that PR was being reviewed, Meta fixed this themselves in https://github.com/facebook/react-native/pull/46970
- C+ (me) didn't do any actual work besides tracking this, so I don't think I'm eligible for any payment
- @harshad2711 is eligible for a reporting bonus it looks like (the issue was reported almost [2 years ago](https://expensify.slack.com/archives/C049HHMV9SM/p1680851400066889))
- @gedu, do you belive you're eligible for any payment?

**Comment by Christinadobrzyn** [comment]
Thanks @eVoloshchak! 

Just a note that I'm going to be ooo from March 28th - April 6th. Back on the 7th. I don't think we'll need anything in the meantime, but if you need a BZ teammate, please reassign the Bug label.

**Comment by eVoloshchak** [comment]
Not overdue

**Comment by Christinadobrzyn** [comment]
Thanks @eVoloshchak!

- Based on this [SO](https://stackoverflowteams.com/c/expensify/questions/11260) the reporting bonus is $250. 
- @gedu can you let me know if need payment for this? 
- @harshad2711 can you please accept this offer - https://www.upwork.com/nx/wm/offer/106843979

**Comment by Christinadobrzyn** [comment]
Update: 

- I paid the $250 reporting bonus to @harshad2711 in Upwork - https://www.upwork.com/nx/wm/offer/106843979
- @gedu are you needing payment for this?

**Comment by Christinadobrzyn** [comment]
nudge @gedu do you need a payment for this or can we close?

**Comment by Christinadobrzyn** [comment]
Closing this out since the other contributors are contractors. Thanks!
